ir/src/symbol_table/mod.rs

```diff
@@ -41,36 +48,36 @@ impl Display for IdentifierType {
     }
 }
 
+// TODO: get rid of need to make this public
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
-pub(super) enum Scope {
+pub(crate) enum Scope {
     BoundaryConstraints,
     IntegrityConstraints,
 }
 
+// TODO: get rid of need to make this public
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq)]
+pub enum VariableValue {
+    Scalar(String),
+    Vector(VectorAccess),
+    Matrix(MatrixAccess),
+}
+
+// SYMBOL TABLE
+// ================================================================================================
+
 /// SymbolTable for identifiers to track their types and information and enforce uniqueness of
 /// identifiers.
 #[derive(Default, Debug)]
 pub struct SymbolTable {
     /// Vector in which index is trace segment and value is number of columns in this segment
     segment_widths: Vec<u16>,
```

Comment:
Looks like this field was also moved to the Declarations struct, should we remove it from here?

Reply:
Great catch! Yes, we should remove it from here and move the `set` handling into the declarations as well. Thank you!

Reply:
I've fixed it. Will you please review the update?